src/Composer/SubprofileMigrationCommand.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Acquia\Lightning\Composer;
+
+use Composer\Composer;
+use Composer\IO\IOInterface;
+use Composer\Script\Event;
+use Drupal\Component\Serialization\Yaml;
+
+/**
+ * Migrates Lightning Extender configuration to a sub-profile.
+ */
+class SubprofileMigrationCommand {
+
+  /**
+   * The IO handler.
+   *
+   * @var \Composer\IO\IOInterface
+   */
+  protected $io;
+
+  /**
+   * The Composer instance.
+   *
+   * @var \Composer\Composer
+   */
+  protected $composer;
+
+  /**
+   * The path to the lightning.extend.yml file, if known.
+   *
+   * @var string|false
+   */
+  protected $path;
+
+  /**
+   * SubprofileMigrationCommand constructor.
+   *
+   * @param \Composer\IO\IOInterface $io
+   *   The IO handler.
+   * @param \Composer\Composer $composer
+   *   The Composer instance.
+   * @param array $arguments
+   *   Arguments passed to the script.
+   */
+  public function __construct(IOInterface $io, Composer $composer, array $arguments = []) {
+    $this->io = $io;
+    $this->composer = $composer;
+    $this->path = reset($arguments);
+  }
+
+  /**
+   * Script entry point.
+   *
+   * @param \Composer\Script\Event $event
+   *   The event that triggered the script.
+   */
+  public static function execute(Event $event) {
+    $command = new static(
+      $event->getIO(),
+      $event->getComposer(),
+      $event->getArguments()
+    );
+    $command->run();
+  }
+
+  /**
+   * Executes the command.
+   */
+  protected function run() {
+    $extender = realpath($this->locateExtender());
+
+    if ($extender) {
+      $extender = file_get_contents($extender);
+      $extender = Yaml::decode($extender);
+
+      // Build the Drupal Console command line.
+      $command = 'lightning:subprofile --no-interaction';
+      $command .= ' --name="Lightning Extender"';
+      $command .= ' --machine-name=lightning_extender';
+
+      if ($extender['modules'] || $extender['lightning_extensions']) {
+        $modules = array_merge(
+          $extender['lightning_extensions'],
+          $extender['modules']
+        );
+        $command .= ' --dependencies="' . implode(', ', $modules) . '"';
+      }
+      if ($extender['exclude_components']) {
+        $command .= ' --exclude-subcomponents="' . implode(', ', $extender['exclude_components']) . '"';
+      }
+
+      // If Drupal Console is installed, go ahead and run the command
+      // non-interactively. Otherwise, echo the command and tell the user run it
+      // themselves.
+      $package = $this->composer
+        ->getRepositoryManager()
+        ->getLocalRepository()
+        ->findPackage('drupal/console', '*');
+
+      if ($package) {
+        $command = $this->composer->getConfig()->get('bin-dir') . '/drupal ' . $command;
+        $this->io->write($command);
```

Comment:
Once things are confirmed working under manual testing we should probably use ProcessExecutor to call Console directly here. That's on me, though.